### PR TITLE
Fix notebook language swapping

### PR DESCRIPTION
**Pull request by ricklamers**
## Description
This PR fixes https://github.com/orchest/orchest/issues/135. In addition, some refactors have been made to support this fix.

Specifically, we've made pipeline changes auto save. Which in turn required upgrading the MDC framework to the latest version.

Credits mainly go to @fruttasecca for his work figuring out the right change kernel JupyterLab requests in order to facilitate this fix.

### Checklist
- [x] The documentation reflects the changes.
- [x] I have manually tested the application to make sure the changes don’t cause any downstream issues.
- [x] In case I changed one of the services’ `models.py` I have performed the appropriate database migrations.

### CODE REVIEW

File: services/orchest-webserver/app/static/js/src/jupyter/Jupyter.js [hunk 1]
```diff
@@ -127,6 +138,29 @@ class Jupyter {
...
         }
+      } else {
+        docManager.services.sessions
+          .findByPath(notebook)
+          .then((notebookSession) => {
+            if (notebookSession) {
+              if (notebookSession.kernel.name !== kernel) {
+                orchest.confirm(
+                  "Warning",
+                  "Do you want to change the active kernel of the opened Notebook? \n\nYou will lose the current kernel's state if no other Notebook is attached to it.",
```
Comment (yannickperrenet) [inline services/orchest-webserver/app/static/js/src/jupyter/Jupyter.js]: I am not entirely sure about the context when this warning is thrown, but wouldn't it be better to say: "Are you sure you want to ..."?

Comment (fruttasecca) [reply services/orchest-webserver/app/static/js/src/jupyter/Jupyter.js]: This is to cover the case of a notebook having a kernel running but the notebook not being open in JupyterLab, thus no widget can be found. About the exact wording, I have no strong preference, @ricklamers ?

Comment (ricklamers) [reply services/orchest-webserver/app/static/js/src/jupyter/Jupyter.js]: What @fruttasecca is saying is correct.

In addition, in the 'regular case' in which JupyterLab is running choosing _not_ to change the active kernel does not revert changing of the kernel in the pipeline definition. As JupyterLab and the pipeline are allowed to diverge if the user explicitly chooses to do so.

**Comment by ricklamers** [comment]
This PR also fixes https://github.com/orchest/orchest/issues/146.

**Pull request merged**